Issue#244

charlie0605 opened the pull request:
Sorting the list when adding Avatars, Sounds, and Backdrop

Avatar:
![image](https://user-images.githubusercontent.com/25602185/30516777-c6fb7f60-9b9c-11e7-9240-4ae4c8280948.png)
![image](https://user-images.githubusercontent.com/25602185/30516789-e98dd442-9b9c-11e7-8696-0d3f950be3e9.png)
![image](https://user-images.githubusercontent.com/25602185/30516790-f20f70e4-9b9c-11e7-98e4-67b1aa7cea8d.png)
![image](https://user-images.githubusercontent.com/25602185/30516791-f8357a2c-9b9c-11e7-9c66-c6903163f27f.png)

Backdrop:
![image](https://user-images.githubusercontent.com/25602185/30516803-50e3cd40-9b9d-11e7-9856-9885c44b3350.png)
![image](https://user-images.githubusercontent.com/25602185/30516806-580ca592-9b9d-11e7-84fc-b4a8b221f439.png)
![image](https://user-images.githubusercontent.com/25602185/30516809-5caddb0c-9b9d-11e7-86b4-c0ba657a1326.png)

Sound:
![image](https://user-images.githubusercontent.com/25602185/30516811-6df90558-9b9d-11e7-89cc-43ea78465c59.png)
![image](https://user-images.githubusercontent.com/25602185/30516812-7c8640fe-9b9d-11e7-9d7c-f1c27fde15bb.png)
![image](https://user-images.githubusercontent.com/25602185/30516814-83744316-9b9d-11e7-837f-867480199486.png)

rohrlach commented on app/views/stages/_avatars.html.erb:
```diff
@@ -3,12 +3,12 @@
   </div>
   <div class="panel-body">
-    <% @stage.avatar_stages.each do |i| %>
+    <% @stage.avatar_stages.sort_by { |x| x.avatar.name }.each do |i| %>
```
Why have you used ```@stage.avatar_stages``` here and ```@stage.backdrops``` on the other template?  Surely ```@stage.avatars``` would be preferable?

rohrlach replied:
Furthermore, on the other template you've opted for ```@stage.sounds.order(:name)``` which seems far more succinct.

thongteav replied:
Hi Paul, I think it was like that before. We only added sort_by to sort the list. So should we modify the code to @stage.avatars instead?

rohrlach replied:
I would say so.  Just because something is, doesn't mean it necessarily should be.  If you see an opportunity to simplify the code - Go for it.

thongteav replied:
Ok.

rohrlach commented:
@charlie0605 can you please elaborate on your intentions with commit cdb75d2?

charlie0605 commented:
@rohrlach 
What I have done is change the avatar controller and model so that the code is consistent with the Backdrops and Sounds.
When I change the code I realised that some of the code is not used, and therefore I have removed the code

While changing the code I realised that when adding an avatar on stage you could always get a success message even if you don't have any Avatar selected.
Also fixed the problem when you try to add tags to stage and avatar it doesn't add it to the stage or avatar, it directs to the new tag created.

thongteav commented:
It seems like Travis test has been passed again.

rohrlach commented:
Thanks @charlie0605 

In general - If you notice extra bugs that are outside the scope of your PR - You should submit a new PR.  PRs should be very specific in what they're designed to achieve.

That being said, I'm happy to accept this one this time provided you comment under the ticket to indicate that your extra work should be tested. I'll accept the changes but wait for you to do this before merging.

charlie0605 commented:
@rohrlach 
Could you please explain it a bit more? Does this mean I have to create another PR on the same issue addressing the issues that were fixed? In this case, I would need to create another PR addressing the assigning avatar to stage button's bug?

rohrlach commented:
@charlie0605 what I'm getting at is that when you submit a PR, you provide a summary of what the PR is intended to provide.  When you add changes to the PR, you need to be careful not to deviate from your stated goals of the PR.  In general, any subsequent commits should be to address test failures and/or review feedback.

charlie0605 commented:
Avatar cannot be assigned with no avatar selected, the button is disabled when no avatars are selected.
![image](https://user-images.githubusercontent.com/25602185/30627742-4b019dea-9e26-11e7-8759-7532cee34d59.png)
when avatars are selected the button is enabled.
![image](https://user-images.githubusercontent.com/25602185/30627764-6f25c228-9e26-11e7-877d-1ea4de31c94b.png)

Tags can now be added to the stage whereas before it will direct you to the tag and not assign to the stage
![image](https://user-images.githubusercontent.com/25602185/30627775-88fcc656-9e26-11e7-9dda-b1934f4a5cb2.png)

Avatars now can be arranged in order of their name
![image](https://user-images.githubusercontent.com/25602185/30627784-a0f714fa-9e26-11e7-97ad-6a37a357cfcf.png)
![image](https://user-images.githubusercontent.com/25602185/30627789-a5824b5c-9e26-11e7-8151-06aafef7a4ac.png)

Avatar tags are now able to be assigned whereas before it directs you to the tag but does not assign it to the avatar.
![image](https://user-images.githubusercontent.com/25602185/30627840-dc4e857e-9e26-11e7-8ec7-b46a083d44bf.png)

rohrlach commented:
Reference #244 to link discussion

rohrlach commented:
User is still able to assign an avatar to a stage with no avatar selected
![image](https://user-images.githubusercontent.com/2830313/30777201-8134255e-a083-11e7-94b8-ee61e5c881f2.png)
![image](https://user-images.githubusercontent.com/2830313/30777206-8f1607b4-a083-11e7-922b-433183ccd488.png)
